### Homework 2

### Question 1: Comparing Two Population Means (Independent t-test)

### Hypotheses

- Null hypothesis ($H_0$): The mean increase in sleep is the same for both drugs, $\mu_1 - \mu_2 = 0$.
- Alternative hypothesis ($H_A$): The mean increase in sleep differs between the drugs, $\mu_1 - \mu_2 \ne 0$.

Because the question asks whether the means differ, this is a two-sided independent-samples t-test. Welch's version is used because it does not assume equal population variances.

The significance level is $\alpha = 0.05$.

In [ ]:
import numpy as np
from scipy import stats


sleep_drug_1 = np.array([0.7, -1.6, -0.2, -1.2, -0.1, 3.4, 3.7, 0.8, 0.0, 2.0])
sleep_drug_2 = np.array([1.9, 0.8, 1.1, 0.1, -0.1, 4.4, 5.5, 1.6, 4.6, 3.4])

alpha = 0.05
result = stats.ttest_ind(sleep_drug_1, sleep_drug_2, equal_var=False)
mean_difference = sleep_drug_1.mean() - sleep_drug_2.mean()

print(f"Drug 1 mean: {sleep_drug_1.mean():.2f} hours")
print(f"Drug 2 mean: {sleep_drug_2.mean():.2f} hours")
print(f"Mean difference (drug 1 - drug 2): {mean_difference:.2f} hours")
print(f"Welch t-statistic: {result.statistic:.3f}")
print(f"Two-sided p-value: {result.pvalue:.4f}")

if result.pvalue < alpha:
    print("Decision: Reject H0. The mean increases differ significantly.")
else:
    print("Decision: Fail to reject H0. There is not enough evidence of a significant difference.")

Drug 1 mean: 0.75 hours
Drug 2 mean: 2.33 hours
Mean difference (drug 1 - drug 2): -1.58 hours
Welch t-statistic: -1.861
Two-sided p-value: 0.0794
Decision: Fail to reject H0. There is not enough evidence of a significant difference.


### Results and conclusion

- The sample mean increase was 0.75 hours for drug 1 and 2.33 hours for drug 2. 
- Thus, drug 2 had a sample mean increase that was 1.58 hours larger than drug 1.

The Welch independent t-test produced $t = -1.861$ and a two-sided $p$-value of 0.0794. Since $p > 0.05$, we fail to reject $H_0$. There is not sufficient statistical evidence at the 5% significance level to conclude that the mean increase in sleep differs between the two drugs.

This conclusion does not prove that the drugs have identical effects; it means that this sample does not provide strong enough evidence of a difference at $\alpha = 0.05$.

### Question 2 ANOVA 
The iris dataset contains measurements for 50 flowers from each of three species of iris: 
setosa, versicolor, and virginica. 

### Context: 
- The iris dataset contains measurements for 50 flowers from each of three species of iris:setosa, versicolor, and virginica. 

### Question: 
- Is there a statistically significant difference in the mean sepal length (Sepal.Length) among the three different iris species (Species)? 
- If so, which groups differ?

In [6]:
import pandas as pd 
import numpy as np 
import seaborn as sns


iris_data = sns.load_dataset('iris')

In [7]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd

alpha = 0.05

# One-way ANOVA: compare mean sepal length across the three species.
groups = [
    iris_data.loc[iris_data["species"] == species, "sepal_length"]
    for species in iris_data["species"].unique()
]
anova_result = stats.f_oneway(*groups)

print("Mean sepal length by species:")
print(iris_data.groupby("species")["sepal_length"].mean().round(3))
print(f"\nANOVA F-statistic: {anova_result.statistic:.3f}")
print(f"ANOVA p-value: {anova_result.pvalue:.6g}")

if anova_result.pvalue < alpha:
    print("\nDecision: Reject H0. At least one species mean differs.")

    tukey_result = pairwise_tukeyhsd(
        endog=iris_data["sepal_length"],
        groups=iris_data["species"],
        alpha=alpha,
    )
    print("\nTukey HSD post-hoc comparisons:")
    print(tukey_result)
else:
    print("\nDecision: Fail to reject H0. There is not enough evidence that the means differ.")

Mean sepal length by species:
species
setosa        5.006
versicolor    5.936
virginica     6.588
Name: sepal_length, dtype: float64

ANOVA F-statistic: 119.265
ANOVA p-value: 1.66967e-31

Decision: Reject H0. At least one species mean differs.

Tukey HSD post-hoc comparisons:
   Multiple Comparison of Means - Tukey HSD, FWER=0.05   
  group1     group2   meandiff p-adj lower  upper  reject
---------------------------------------------------------
    setosa versicolor     0.93   0.0 0.6862 1.1738   True
    setosa  virginica    1.582   0.0 1.3382 1.8258   True
versicolor  virginica    0.652   0.0 0.4082 0.8958   True
---------------------------------------------------------


### Results and conclusion

The mean sepal lengths are:

- **Setosa:** 5.006
- **Versicolor:** 5.936
- **Virginica:** 6.588

The one-way ANOVA produced $F(2,147) = 119.265$ with $p \approx 1.67 \times 10^{-31}$. Because $p < 0.05$, we reject $H_0$ and conclude that mean sepal length is not the same for all three iris species.

Tukey's HSD post-hoc test found that every pair differs significantly at the 5% level:

- Setosa has a significantly shorter mean sepal length than versicolor.
- Setosa has a significantly shorter mean sepal length than virginica.
- Versicolor has a significantly shorter mean sepal length than virginica.

Therefore, all three species differ significantly in mean sepal length, with virginica having the largest sample mean and setosa the smallest.